Library/Homebrew/cli_parser.rb

```diff
@@ -82,6 +92,90 @@ def common_switch(name)
         else name
         end
       end
+
+      def option_passed?(name)
+        @parsed_args.respond_to?(name) || @parsed_args.respond_to?("#{name}?")
+      end
+
+      def set_constraints(name, depends_on:, required_for:)
+        secondary = option_to_name(name)
+        unless required_for.nil?
+          primary = option_to_name(required_for)
+          @constraints << [primary, secondary, :mandatory]
+        end
+
+        return if depends_on.nil?
+        primary = option_to_name(depends_on)
+        @constraints << [primary, secondary, :optional]
+      end
+
+      def check_constraints
+        @constraints.each do |primary, secondary, constraint_type|
+          primary_passed = option_passed?(primary)
+          secondary_passed = option_passed?(secondary)
+          if :mandatory.equal?(constraint_type) && primary_passed && !secondary_passed
+            raise OptionConstraintError.new(primary, secondary)
+          end
+          raise OptionConstraintError.new(primary, secondary, missing: true) if secondary_passed &&
```

Comment:
Make this an `elsif`?

Reply:
This leads to RuboCop style error:
> Use a guard clause instead of wrapping the code inside a conditional expression.

Reply:
Instead could you put the `if secondary_passed ...` on it's own line (rather than a suffix)

Reply:
This works, Updated the code.